### ISSUE TITLE
Polyline edge waypoints will not be removed

### ISSUE THREAD

**Issue by ivy-lli** [eclipse-glsp/glsp#612]
If you have an polyline edge and you add a waypoint, there is no way to remove this waypoint again. Even if the feedback shows that the waypoint will be removed (the 2 dots on the right and left of the moved handle disappears):

![glsp-waypoint-bug](https://user-images.githubusercontent.com/42733123/162436361-f2a87ae8-9e74-491a-b337-1b46b605916e.gif)

If you try something similar in the plain sprotty demo, it seems to work:

![sprotty-waypoints](https://user-images.githubusercontent.com/42733123/162436544-8318a3fd-c6f2-42ee-8a76-9c7eb9b066b1.gif)

**Comment by rsoika** [comment]
I have observed similar behaviour with sportty Manhattan router too. After I [uprated to the latest sprotty version](https://github.com/eclipse-glsp/glsp/discussions/611) the routing shows a new and now much better behaviour. But I can not say if this is also true for the Polyline Router. 

Anyway, I have a different issue with moving elements and recomputing the route. This goes in a similar direction I think. I have opened in sprotty the issue: https://github.com/eclipse/sprotty/issues/284

**Comment by gabrieldulca** [comment]
I have looked into this and it appears that on `mouseUp()` the `EdgeEditListener` creates a `ChangeRoutingPointsOperation` that also contains the routing point(s) that should have been removed. This behavior seems to be inconsistent with the one present in the `FeedbackEdgeRouteMovingMouseListener`. 

Additionally I have also noticed that dragging a routing point only works once and the edge will jump back to its previous route upon releasing the mouse click. Deselecting the edge will reset this behavior and allow a routing point to be dragged one more time. 

![Routing bug](https://user-images.githubusercontent.com/16562670/185108104-17813d71-00e4-4b0d-a58f-e5a978be4440.gif)
